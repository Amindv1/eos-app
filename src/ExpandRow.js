import {BootstrapTable, TableHeaderColumn} from 'react-bootstrap-table';
import React from 'react';
// import BSTable from './BSTable'

class BSTable extends React.Component {
  render() {
    if (this.props.data) {
      let style = { "tableLayout": "fixed", "wordWrap": "breakWord", whiteSpace: 'normal' }
      return (
        <BootstrapTable data={ this.props.data }>
          <TableHeaderColumn tdStyle={style} thStyle={style} dataField='raw' isKey={true}>Raw data</TableHeaderColumn>
        </BootstrapTable>);
    } else {
      return (<p>?</p>);
    }
  }
}

class ExpandRow extends React.Component {
  isExpandableRow(row) {
    return true;
  }

  expandComponent(row) {
    let str = JSON.stringify(row.raw);
    let newData = [{raw: str}];

    return (
      <BSTable data={ newData } />
    );
  }

  render() {
    const options = {
      expandRowBgColor: 'rgb(242, 255, 163)',
    };

    return (
      <BootstrapTable data={ this.props.data }
        options={ options }
        expandableRow={ this.isExpandableRow }
        expandComponent={ this.expandComponent }
        search>
        <TableHeaderColumn dataField='hash' isKey={ true }>Block Timestamp Hash</TableHeaderColumn>
        <TableHeaderColumn dataField='action' >Action</TableHeaderColumn>
      </BootstrapTable>
    );
  }
}

export default ExpandRow;