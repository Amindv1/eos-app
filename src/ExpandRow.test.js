import React from 'react';
import ReactDOM from 'react-dom';
import ExpandRow from './ExpandRow';
import toJson from 'enzyme-to-json';
import { shallow, mount, render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {BootstrapTable, TableHeaderColumn} from 'react-bootstrap-table';

configure({ adapter: new Adapter() });

it('Expand row renderes without crashing', () => {
  const block = [{}]
  const data = {block}

  const component = shallow(<ExpandRow data={data.block}/>);
  expect(toJson(component)).toMatchSnapshot();
});

it('bootstrap table present', () => {
  const block = [{
    hash: 1,
    raw: "some string",
    action: 0
  },
  {
    hash: 2,
    raw: "some string",
    action: 0
  },
  {
    hash: 3,
    raw: "some string",
    action: 0
  }]
  const data = {block}

  const wrapper = shallow(<ExpandRow data={data.block}/>);
  expect(wrapper.find(BootstrapTable).length).toBe(1);
  expect(wrapper.find(TableHeaderColumn).length).toBe(2);
});

it('expand row shows raw data', () => {
  const block = [{
    hash: 1,
    raw: "rawdata",
    action: 0
  }]
  const data = {block}
  const wrapper = mount((
    <ExpandRow data={data.block}/>
  ));
  expect(wrapper.find('ExpandComponent').props().hidden).toBe(true)
  expect(wrapper.find('tr').length).toBe(5);
  expect(wrapper.find('td[children="1"]').length).toBe(1);
  expect(toJson(wrapper)).toMatchSnapshot();

  wrapper.find('td[children="1"]').simulate('click');
  
  expect(wrapper.find('ExpandComponent').props().hidden).toBe(false)
  expect(toJson(wrapper)).toMatchSnapshot();
});
