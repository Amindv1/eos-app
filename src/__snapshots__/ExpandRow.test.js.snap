// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expand row renderes without crashing 1`] = `
<BootstrapTable
  autoCollapse={
    Object {
      "filter": false,
      "search": false,
      "sort": false,
    }
  }
  bodyContainerClass={null}
  bordered={true}
  cellEdit={
    Object {
      "afterSaveCell": undefined,
      "beforeSaveCell": undefined,
      "blurToEscape": false,
      "blurToSave": false,
      "mode": "none",
      "nonEditableRows": undefined,
    }
  }
  columnFilter={false}
  condensed={false}
  containerClass={null}
  csvFileName="spreadsheet.csv"
  data={
    Array [
      Object {},
    ]
  }
  deleteRow={false}
  expandColumnOptions={
    Object {
      "expandColumnBeforeSelectColumn": true,
      "expandColumnComponent": undefined,
      "expandColumnVisible": false,
      "expandedColumnHeaderComponent": undefined,
    }
  }
  expandComponent={[Function]}
  expandableRow={[Function]}
  exportCSV={false}
  fetchInfo={
    Object {
      "dataTotalSize": 0,
    }
  }
  footer={false}
  headerContainerClass={null}
  height="100%"
  hover={false}
  ignoreSinglePage={false}
  insertRow={false}
  keyBoardNav={false}
  multiColumnSearch={false}
  multiColumnSort={1}
  options={
    Object {
      "expandRowBgColor": "rgb(242, 255, 163)",
    }
  }
  pagination={false}
  printable={false}
  renderAlert={true}
  replace={false}
  search={true}
  selectRow={
    Object {
      "bgColor": "",
      "clickToExpand": false,
      "clickToSelect": false,
      "clickToSelectAndEditCell": false,
      "customComponent": undefined,
      "hideSelectColumn": false,
      "mode": "none",
      "onSelect": undefined,
      "onSelectAll": undefined,
      "onlyUnselectVisible": false,
      "selected": Array [],
      "showOnlySelected": false,
      "unselectable": Array [],
    }
  }
  striped={false}
  tableBodyClass={null}
  tableContainerClass={null}
  tableFooterClass={null}
  tableHeaderClass={null}
  trClassName=""
  version="3"
  withoutTabIndex={false}
>
  <TableHeaderColumn
    className=""
    columnClassName=""
    columnTitle={false}
    csvFieldType="string"
    dataAlign="left"
    dataField="hash"
    dataSort={false}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={true}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}
  >
    Block Timestamp Hash
  </TableHeaderColumn>
  <TableHeaderColumn
    className=""
    columnClassName=""
    columnTitle={false}
    csvFieldType="string"
    dataAlign="left"
    dataField="action"
    dataSort={false}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}
  >
    Action
  </TableHeaderColumn>
</BootstrapTable>
`;
